### ISSUE TITLE
Teradata CREATE TABLE is marked as unparsabe section.

### ISSUE THREAD

**Issue by nervocalm** [sqlfluff/sqlfluff#6700]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

A simple create multiset table is marked as unparsable section.

am I missing something?

Thanks.

### Expected Behaviour

To parse the code so can later be reformatted

### Observed Behaviour

> sqlfluff lint test.sql --dialect teradata
> 
> == [test.sql] FAIL 
> L:  65 | P:   1 |  PRS | Line 65, Position 1: Found unparsable section: 'CREATE
>                        | MULTISET TABLE TEST.TEST_01, fALLBACK, NO ...'
> WARNING: Parsing errors found and dialect is set to 'teradata'. Have you configured your dialect correctly?
> 
Even the simplest code, return this error:

```
CREATE TABLE TEST.TEST_01, FALLBACK, NO BEFORE JOURNAL, CHECKSUM = DEFAULT, DEFAULT MERGEBLOCKRATIO, MAP = TD_MAP1 (
test_field_01 INTEGER,
test_field_02 VARCHAR(50) CHARACTER SET LATIN NOT CASESPECIFIC
) PRIMARY INDEX (test_field_01);
```

### How to reproduce

1.  Write a simple create table:

```
CREATE TABLE TEST.TEST_01, FALLBACK, NO BEFORE JOURNAL, CHECKSUM = DEFAULT, DEFAULT MERGEBLOCKRATIO, MAP = TD_MAP1 (
test_field_01 INTEGER,
test_field_02 VARCHAR(50) CHARACTER SET LATIN NOT CASESPECIFIC
) PRIMARY INDEX (test_field_01);
```
2.  run
> sqlfluff lint test.sql --dialect teradata

### Dialect

teradata

### Version

3.3.0-1

### Configuration

don't have one. Is running by default.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by daniepi** [comment]
Hi. Isn't `CREATE TABLE` or `CREATE MULTISET TABLE` or `VOLATILE` a standard Teradata syntax. Till now we have had to comment out those parts of the code for sqlfluff to be able to lint the SQL.

**Comment by peterbud** [comment]
With the latest `sqlfluff` all above examples are parsing and fixed properly to this:
```sql
CREATE MULTISET TABLE TEST.TEST_01,
FALLBACK,
NO BEFORE JOURNAL,
CHECKSUM = DEFAULT,
DEFAULT MERGEBLOCKRATIO,
MAP = TD_MAP1 (
    TEST_FIELD_01 INTEGER,
    TEST_FIELD_02 VARCHAR(50) CHARACTER SET LATIN NOT CASESPECIFIC
) PRIMARY INDEX (TEST_FIELD_01);
```
